### PR TITLE
AsyncButton: protect against double submit

### PR DESCRIPTION
**Pull request by matildepark**
Supersedes #4070, #3877, don't merge until the changes are in indigo-react. Re-opened after #4070 auto-closed.

### CODE REVIEW

**Comment by matildepark** [comment]
@liam-fitzgerald This is unblocked by indigo-react release.

**Comment by liam-fitzgerald** [comment]
ready to go!

File: pkg/interface/package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
     "@reach/tabs": "^0.10.5",
     "@tlon/indigo-dark": "^1.0.6",
     "@tlon/indigo-light": "^1.0.6",
-    "@tlon/indigo-react": "1.2.16",
+    "@tlon/indigo-react": "1.2.17",
```
Comment (matildepark) [inline pkg/interface/package.json:13]: Surprised this works when 1.2.17 isn't released.

Comment (matildepark) [reply pkg/interface/package.json:13]: Actually [I take that back](https://www.npmjs.com/package/@tlon/indigo-react), the GitHub and NPM have gotten out of sync.

**Review by tacryt-socryp — APPROVED** [review]

**Pull request merged**